### PR TITLE
[Fix] Invalid password error 

### PR DESCRIPTION
**Pull request by owencraston**
**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

**Description**

_Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions,_
_1. What is the reason for the change?_
- so users dont get stuck in a state where they have an invalid password when they are forced to backup
_2. What is the improvement/solution?_
- set the global variable of passwordSet to true when a user goes through vault corruption

#### Android
- Install Apk build 1127 v7.1.0
- Import account using SRP
- Navigate to main wallet view and kill app
- Install Apk build 1129 v7.0.2
- Open app and type in password

#### iOS
- Install v7.1.0 (1127) on IOS device using TestFlight
- Create wallet and navigate to main wallet view
- Kill app, and update to v7.0.2 (1129) using TestFlight
- Relaunch app, input password on login page
- Go through the vault recovery and navigate to main wallet view
- Kill app, then relaunch app

**Screenshots/Recordings**

_If applicable, add screenshots and/or recordings to visualize the before and after of your change_

**Issue**

Progresses https://github.com/MetaMask/mobile-planning/issues/975

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

**Review by Andepande — CHANGES_REQUESTED** [review]
Issue 1 [Android Specific]
Details
- When user imports from SRP, and updates from build 1127 to 1129, user enters valid password, an 'Invalid Password' validation error occurs on login page 

Steps to Reproduce:

1. Install Apk build 1127 v7.1.0
2. Import account using SRP
3. Navigate to main wallet view and kill app
4. Install Apk build 1129 v7.0.2
5. Open app and type in password

Outcome 

- Invalid Password validation error occurs 

Device: Android 13 Pixel 6,
see here for android, skip to minute 2, error occurs at 2:48 https://recordit.co/m8gfHFhbDz

Issue 2 [IOS specific]
- Vault corruption flow is stuck in a loop. When the user kills and relaunches the app, user is taken through the vault recovery flow again repeatedly.
Device:  IOS 16 iPhone SE

Steps to reproduce:
1. Install v7.1.0 (1127) on IOS device using TestFlight
2. Create wallet and navigate to main wallet view
3. Kill app, and update to v7.0.2 (1129) using TestFlight
4. Relaunch app, input password on login page
5. Go through the vault recovery and navigate to main wallet view 
6. Kill app, then relaunch app

Outcome:
- Vault Recovery Process repeats again

There are two recordings, the first [recording](https://recordit.co/oXAFX5rvrc) which goes through the upgrading from v7.1.0 (1127) to v7.0.2 (1129).
The second [recording](https://recordit.co/Wa5qn3omSL) is a continuation (Recording stopped accidentally), this illustrates the Vault recovery loop when the app is killed on IOS


**Clarification on expected behavior:**
1. After a user who imported their wallet goes through the vault recovery flow, should the protect your wallet modal appear when a user already imported their wallet via SRP?
see recording: http://recordit.co/Mz2zObw2gO
The protect your wallet modal does not appear when a user goes through the vault recovery flow after a new wallet was created: http://recordit.co/DH9cDN0kgE

2. When upgrading the app between versions, the vault recovery flow is triggered. Is this now expected behavior? This seems a bit counterintuitive because we are now prompting all users to recover their wallet once they upgrade between app versions.

**Comment by sethkfman** [comment]
> When upgrading the app between versions, the vault recovery flow is triggered. Is this now expected behavior? This seems a bit counterintuitive because we are now prompting all users to recover their wallet once they upgrade between app versions.

This is not expected behavior and would be part of the blocking issue. @Andepande @owencraston

**Comment by sethkfman** [comment]
This probably should go into the next release but is not the root cause of the password issue. @owencraston

File: app/components/Views/Login/index.js [hunk 1]
```diff
@@ -381,11 +381,13 @@ class Login extends PureComponent {
...
-        authType,
-        this.props.selectedAddress,
-      );
+      // await Authentication.userEntryAuth(
+      //   password,
+      //   authType,
+      //   this.props.selectedAddress,
+      // );
+
+      throw new Error(VAULT_ERROR);
```
Comment (owencraston) [inline app/components/Views/Login/index.js:390]: this is only for manually triggering a vault corruption error so we can go through the recovery flow.

**Pull request closed without merging**